File: 02 - Compute the Longest Common Prefix/solution.js
function longestCommonPrefix(strs) {
  if (strs.length === 0) return "";

  let prefix = strs[0];

  for (let i = 1; i < strs.length; i++) {
    while (strs[i].indexOf(prefix) !== 0) {
      prefix = prefix.substring(0, prefix.length - 1);

      if (prefix.length === 0) return "";
    }
  }

  return prefix;
}

// Example usage
console.log(longestCommonPrefix(["flower", "flow", "flight"])); // Output: "fl"
console.log(longestCommonPrefix(["dog", "racecar", "car"])); // Output: ""
console.log(longestCommonPrefix(["interstellar", "internet", "internal"])); // Output: "inte"
console.log(longestCommonPrefix(["apple", "ape", "april"])); // Output: "ap"

File: 02 - Compute the Longest Common Prefix/solution.py
def  longest_common_prefix(strs):
    if len(strs) == 0:
        return ''
    
    prefix = strs[0]

    for i in range (1, len(strs)):
        while not strs[i].startswith(prefix):
            prefix = prefix[:-1]
            if len(prefix) == 0:
                return ''
    
    return prefix





# Example usage
print(longest_common_prefix(["flower", "flow", "flight"]))  # Output: "fl"
print(longest_common_prefix(["dog", "racecar", "car"]))  # Output: ""
print(longest_common_prefix(["interstellar", "internet", "internal"]))  # Output: "inte"
print(longest_common_prefix(["apple", "ape", "april"]))  # Output: "ap"
